images/build/Dockerfiles/vso.bullseye.Dockerfile

```diff
@@ -0,0 +1,325 @@
+        git \
+        make \
+        unzip \
+        # The tools in this package are used when installing packages for Python
+        build-essential \
+        # swig3.0 \
+        # Required for Microsoft SQL Server
+        unixodbc-dev \
+        # Required for PostgreSQL
+        libpq-dev \
+        # Required for mysqlclient
+        default-libmysqlclient-dev \
+        # Required for ts
+        moreutils \
+        rsync \
+        zip \
+        libgdiplus \
+        jq \
+        # By default pip is not available in the buildpacks image
+        python-pip-whl \
+        python3-pip \
+        #.NET Core related pre-requisites
+        libc6 \
+        libgcc1 \
+        libgssapi-krb5-2 \
+        libncurses5 \
+        liblttng-ust0 \
+        libssl-dev \
+        libstdc++6 \
+        zlib1g \
+        libuuid1 \
+        libunwind8 \
+        sqlite3 \
+        libsqlite3-dev \
+        software-properties-common \
+    && rm -rf /var/lib/apt/lists/* \
+    # This is the folder containing 'links' to benv and build script generator
+    && apt-get update \
+    && apt-get upgrade -y \
+    # && add-apt-repository universe \
+    && apt-get install -y --no-install-recommends python2 \
+    && rm -rf /var/lib/apt/lists/* \
+    # 'get-pip.py' has been moved to ' https://bootstrap.pypa.io/pip/2.7/get-pip.py' from 'https://bootstrap.pypa.io/2.7/get-pip.py'
+    #&& curl  https://bootstrap.pypa.io/pip/2.7/get-pip.py --output get-pip.py \
+    #&& python2 get-pip.py \
+    #&& pip install pip --upgrade \
+    && pip3 install pip --upgrade \
+    && mkdir -p /opt/oryx
+
+# NOTE: Place only folders whose size does not extrememly effect the perf of building this image
+# since this intermediate stage is copied to final stage.
+# For example, if we put yarn-cache here it is going to impact perf since it more than 500MB
+FROM main AS intermediate
+COPY --from=oryxdevmcr.azurecr.io/private/oryx/support-files-image-for-build /tmp/oryx/ /opt/tmp
+COPY --from=oryxdevmcr.azurecr.io/private/oryx/buildscriptgenerator /opt/buildscriptgen/ /opt/buildscriptgen/
+ 
+FROM main AS final
+ARG AI_KEY
+ARG SDK_STORAGE_BASE_URL_VALUE
+
+# add an environment variable to determine debian_flavor
+# to correctly download platform sdk during platform installation
+ENV DEBIAN_FLAVOR="bullseye"
+
+COPY --from=intermediate /opt /opt
+
+# Docker has an issue with variable expansion when all are used in a single ENV command.
+# For example here the $LASTNAME in the following example does not expand to JORDAN but instead is empty: 
+#   ENV LASTNAME="JORDAN" \
+#       NAME="MICHAEL $LASTNAME"
+#
+# Even though this adds a new docker layer we are doing this 
+# because we want to avoid duplication (which is always error-prone)
+
+RUN set -ex \
+    && tmpDir="/opt/tmp" \
+    && imagesDir="$tmpDir/images" \
+    && buildDir="$tmpDir/build" \
+    # https://github.com/docker-library/python/issues/147
+    && PYTHONIOENCODING="UTF-8" \    
+    # new next line
```

Comment:
I see this comment a few times in this file now -- are we able to remove it or is it needed for documentation purposes? I'm in the boat that if we want to keep a comment on this line, it should explain what's going on with the `add-apt-repository` command called below it.

Reply:
Thanks for pointing this out, this line should no longer be present.